core/services/job/orm.go

```diff
@@ -180,14 +181,37 @@ func (o *orm) CreateJob(jb *Job, qopts ...pg.QOpt) error {
 				}
 			}
 
+			existingSpec := new(OCROracleSpec)
+			err := tx.Get(existingSpec, `SELECT * FROM ocr_oracle_specs WHERE contract_address = $1 and (evm_chain_id = $2 or evm_chain_id IS NULL) LIMIT 1;`,
```

Comment:
should be able to just detect that the unique constraint was violated and wrap the error with more info instead of this extra query

Reply:
the query would need to happen anyways, since we need to validate that if the spec exists but it has null as the evm_chain_id, that the evm_chain_id of the new spec it is not the default chain id. 

This is because the one in the database will already default to the default chain, and we have a client that ran into an error in which they were able to add 2 OCR jobs for the same contract address and the node failed.